webpaste.el

```diff
@@ -300,6 +307,22 @@ This is the default failover hook that we use for most providers."
                    ;; Otherwise we return the formatted post data
                    post-data))))
 
+(cl-defun webpaste--providers-pinnwand-request (&key text post-data provider-uri
+                                                  &allow-other-keys)
+  "Build request for pinnwand pastebins."
+  (let* ((lexer (or (webpaste--get-buffer-language provider-uri) "text"))
+         (file `(("lexer" . ,lexer) ("content" . ,text)))
+         (file-name (buffer-file-name)))
+    (when file-name
+      (push (cons "name" (file-name-nondirectory file-name)) file))
+    (json-encode `((expiry . ,(or (cdr (assoc "expiry" post-data)) "1day"))
+                   (files . ,(vector file))))))
```

Comment:
Se here we're wrapping a lambda pretty much instead of in the provider declaration.
```suggestion
(cl-defun webpaste--providers-pinnwand-request ()
  "Build request for pinnwand pastebins."
  (cl-function (lambda (&key text post-data provider-uri &allow-other-keys)
                 "Build request for pinnwand pastebins."
                 (let* ((lexer (or (webpaste--get-buffer-language provider-uri) "text"))
                        (file `(("lexer" . ,lexer) ("content" . ,text)))
                        (file-name (buffer-file-name)))
                   (when file-name
                     (push (cons "name" (file-name-nondirectory file-name)) file))
                   (json-encode `((expiry . ,(or (cdr (assoc "expiry" post-data)) "1day"))
                                  (files . ,(vector file))))))))

```